python/tank/descriptor/io_descriptor/base.py

```diff
@@ -693,6 +702,26 @@ def clone_cache(self, cache_root):
         filesystem.copy_folder(source_cache_path, new_cache_path, skip_list=[])
         return True
 
+    def attempt_move(self, src, dst):
+        """
+        Attempts to move a file/folder from source to destination.
+        :param src: The source file/folder.
+        :param dst: The destination file/folder.
+        """
+        # Proceed with the move only if the destination is not an existing folder.
+        # If the destination is a folder and exists, the move operation causes the source
+        # to be nested inside the destination folder, which is not the desirable behavior.
+        try:
+            if not os.path.isdir(dst):
```

Comment:
I agree with you @jfboismenu, I would have liked it to be an atomic move statement too if possible. But if not for an existence check, how should it work around the default move command behavior of nesting folders if the destination folder already exists ? I added a comment above about that just in case i.e. > `mv src dst` resulting in `dst/src`. If the destination folder exists, it tends to simply create the folder with that folder without raising an exception. 

I agree with the need for an additional check for the OSError (as it could it a permission error as well).

Reply:
Sorry, seems like I read your comment too quickly. How about [os.rename](https://docs.python.org/2/library/os.html#os.rename) then? The documentation says that if the destination exist it will fail.